cfme/tests/infrastructure/test_host.py

```diff
@@ -540,26 +541,51 @@ def test_add_ipmi_refresh(appliance, setup_provider):
 
 
 @test_requirements.infra_hosts
-@pytest.mark.parametrize("crud_action", ['edit_from_hosts', 'edit_from_details', 'remove'])
+@pytest.mark.meta(automates=[1634794])
+@pytest.mark.parametrize("crud_action", ['edit_from_hosts', 'edit_from_details', 'cancel',
+                                         'nav_away_changes', 'nav_away_no_changes', 'remove'])
 def test_infrastructure_hosts_crud(appliance, setup_provider, crud_action):
     """
     Polarion:
         assignee: prichard
         casecomponent: Infra
         caseimportance: low
         initialEstimate: 1/6h
+    Bugzilla:
+        1634794
     """
     host = appliance.collections.hosts.all()[0]
+    if crud_action in ['cancel', 'nav_away_changes', 'nav_away_no_changes']:
```

Comment:
The if/else logic is hard to follow:

```
    if crud_action in ['cancel', 'nav_away_changes', 'nav_away_no_changes']:

    if crud_action != 'remove':

        if crud_action not in ['cancel', 'nav_away_changes', 'nav_away_no_changes']:

        else:

    else:
```

I think something like the following would be clearer:

```
if crud_action == 'remove':
    [delete() calls here]

elif crud_action in ('edit_from_hosts', 'edit_from_details'):
    [update() call here]

else:
    [navigate_away/cancel helper method call here]
```

Reply:
I added comments and updated the structure of test_infrastructure_hosts_crud to make things clearer. Still adding comments where I think they would help clarify.

Reply:
I added comments and rearranged the structure of test_infrastructure_hosts_crud to make things clearer. I am still adding commants to host.update